Promact.Oauth.Server/src/Promact.Oauth.Server/Controllers/ProjectController.cs

```diff
@@ -44,356 +44,500 @@ public class ProjectController : Controller
+        * @apiError BadRequest
+        * @apiErrorExample {json} Error-Response:
+        * HTTP/1.1 404 ProjectNotFound Project Not Found
+        * {
+        *   "error": "ProjectNotFound"
+        * }    
         */
         [Authorize]
         [HttpPut]
-        [Route("editProject")]
-        public async Task<IActionResult> editProject(int id, [FromBody]ProjectAc project)
+        [Route("{id:int}")]
+        public async Task<IActionResult> EditProjectAsync(int id,[FromBody]ProjectAc project)
         {
             try
             {
                 var updatedBy = _userManager.GetUserId(User);
-
                 if (ModelState.IsValid)
                 {
-                    ProjectAc projectAc = _projectRepository.checkDuplicateFromEditProject(project);
-                    if (projectAc.Name != null && projectAc.SlackChannelName != null)
+                    ProjectAc projectAc = await _projectRepository.CheckDuplicateProjectAsync(project);
+                    if (!string.IsNullOrEmpty(projectAc.Name) && !string.IsNullOrEmpty(projectAc.SlackChannelName))
                     {
-                        await _projectRepository.EditProject(project, updatedBy);
+                        await _projectRepository.EditProjectAsync(id, project, updatedBy);
                     }
                     else { return Ok(project); }
                 }
                 return Ok(project);
             }
-            catch (Exception ex)
+            catch (ProjectNotFound)
             {
-                ex.ToExceptionless().Submit();
-                throw ex;
+                return NotFound();
             }
         }
 
         /**
-        * @api {get} api/Project/fetchProject 
+        * @api {get} api/project/:name 
         * @apiVersion 1.0.0
-        * @apiName Project
+        * @apiName GetProjectByGroupNameAsync
         * @apiGroup Project
         * @apiParam {string} name project Name
         * @apiParamExample {json} Request-Example:
-        *      
-        *        {
-        *             
-        *             "Name":"ProjectName"
-        *            
-        *        }      
+        * {
+        *   "Name":"ProjectName"
+        * }      
         * @apiSuccessExample {json} Success-Response:
         * HTTP/1.1 200 OK 
         * {
-        *     "description":"Object of ProjectAc"
+        *    
+        *   "Name":"ProjectName",
+        *   "SlackChannelName":"SlackChannelName",
+        *   "IsActive":"True",
+        *   "TeamLeaderId":"1",
+        *   "ApplicationUsers":null
+        * }
+        * @apiError ProjectNotFound The id of the Project was not found.
+        * @apiErrorExample {json} Error-Response:
+        * HTTP/1.1 404 Not Found
+        * {
+        *   "error": "ProjectNotFound"
         * }
         */
         [ServiceFilter(typeof(CustomAttribute))]
         [HttpGet]
-        [Route("fetchProject/{name}")]
-        public ProjectAc Fetch(string name)
+        [Route("{name}")]
```

Comment:
using route constraints. in project module project id is int and name is string. so we can separate it